src/store/reducer.js

```diff
@@ -0,0 +1,78 @@
+import * as actions from '../consts/types';
+
+const initialState = {
+    cart: [],
+    totalPrice: 0
+};
+
+const addProduct = (state, action) => {
+    const newProduct = {
+        id: action.product.id, name: action.product.name, price: action.product.price, description: action.product.description,
+        brand: { id: action.product.brand.id, name: action.product.brand.name }, size: action.product.size, camera: action.product.camera,
+        cpu: action.product.cpu, memory: action.product.memory, display: action.product.display, battery: action.product.battery, imageUrl: action.product.imageUrl, amount: 1
+    };
+    let newCart = [...state.cart];
+    const addedPrice = newProduct.price * newProduct.amount;
+    newCart.push(newProduct);
+    return { ...state, cart: newCart, totalPrice: state.totalPrice + addedPrice };
+}
+
+const removeProduct = (state, action) => {
+    let filteredCart = [...state.cart];
```

Comment:
This is and in line 39 are the only places here you realy need a let, in all of the rest you can use const.
You can change a const json's field value or push to const array.